tests/rule-grouping/rule-grouping-4/test.yaml

```diff
@@ -0,0 +1,151 @@
+requires:
+  min-version: 8
+
+pcap: false
+
+args:
+  - --engine-analysis
+
+checks:
+  - filter:
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[0].port: 7
+        tcp.toserver[0].port2: 7
+        tcp.toserver[0].rulegroup.id: 0
+        tcp.toserver[0].rulegroup.rules[0].sig_id: 1
+        tcp.toserver[0].rulegroup.rules[1].sig_id: 2
+        tcp.toserver[0].rulegroup.rules[2].sig_id: 3
+        tcp.toserver[0].rulegroup.rules[3].sig_id: 6
+  - filter:
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[1].port: 8
+        tcp.toserver[1].port2: 8
+        tcp.toserver[1].rulegroup.id: 0
+        tcp.toserver[1].rulegroup.rules[0].sig_id: 1
+        tcp.toserver[1].rulegroup.rules[1].sig_id: 2
+        tcp.toserver[1].rulegroup.rules[2].sig_id: 3
+        tcp.toserver[1].rulegroup.rules[3].sig_id: 6
+  - filter:
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[2].port: 25
+        tcp.toserver[2].port2: 42
+        tcp.toserver[2].rulegroup.id: 1
+        tcp.toserver[2].rulegroup.rules[0].sig_id: 2
+        tcp.toserver[2].rulegroup.rules[1].sig_id: 3
+        tcp.toserver[2].rulegroup.rules[2].sig_id: 5
+        tcp.toserver[2].rulegroup.rules[3].sig_id: 6
```

Comment:
this is a bit odd. 6 is 7-25, so it includes 25. This group is about 25-42. Should there be a group 25-25 and then 26-42?

Reply:
Indeed. That's how it should have been as per the algo 🤔 so, there's a bug. Thank you very much for the careful review